### PR TITLE
Implement GetAllUsersSlots on CoordinatorController

### PR DESCRIPTION
**Pull request by Famezzs**

### CODE REVIEW

**Comment by knurff** [comment]
|File|Coverage [21%]|:green_apple:|
|:-|:-:|:-:|
|[DateUtils.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2FDateUtils.java)|67.82%|:green_apple:|
|[CandidateService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateService.java)|57.14%|:green_apple:|
|[CoordinatorService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCoordinatorService.java)|38.71%|:green_apple:|
|[InterviewerSlot.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2FInterviewerSlot.java)|21.48%|:green_apple:|
|[DashboardDayDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDayDto.java)|13.82%|:green_apple:|
|[DashboardDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDto.java)|9.13%|:x:|
|[InterviewerService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FInterviewerService.java)|0%|:x:|
|[BookingService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FBookingService.java)|0%|:x:|
|[CoordinatorController.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/2049c24d21bcec370fac193d8b211658c1c15110/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2FCoordinatorController.java)|0%|:x:|

|Total Project Coverage|16.79%|:green_apple:|
|:-|:-:|:-:|

**Comment by Famezzs** [comment]
Currently working on adding more tests

**Comment by knurff** [comment]
|File|Coverage [21%]|:green_apple:|
|:-|:-:|:-:|
|[DateUtils.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2FDateUtils.java)|67.82%|:green_apple:|
|[CandidateService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateService.java)|57.14%|:green_apple:|
|[CoordinatorService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCoordinatorService.java)|38.71%|:green_apple:|
|[InterviewerSlot.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2FInterviewerSlot.java)|21.48%|:green_apple:|
|[DashboardDayDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDayDto.java)|13.82%|:green_apple:|
|[DashboardDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDto.java)|9.13%|:x:|
|[InterviewerService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FInterviewerService.java)|0%|:x:|
|[BookingService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FBookingService.java)|0%|:x:|
|[CoordinatorController.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/b05f46def7bf0773da3cbd0b2175b4c814ba4ddb/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2FCoordinatorController.java)|0%|:x:|

|Total Project Coverage|16.79%|:green_apple:|
|:-|:-:|:-:|

**Comment by knurff** [comment]
|File|Coverage [30.01%]|:green_apple:|
|:-|:-:|:-:|
|[BookingService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FBookingService.java)|100%|:green_apple:|
|[DateUtils.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2FDateUtils.java)|67.82%|:green_apple:|
|[CandidateService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateService.java)|57.14%|:green_apple:|
|[CoordinatorService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCoordinatorService.java)|38.71%|:green_apple:|
|[InterviewerSlot.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2FInterviewerSlot.java)|21.48%|:green_apple:|
|[DashboardDayDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDayDto.java)|13.82%|:green_apple:|
|[DashboardDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDto.java)|9.13%|:x:|
|[InterviewerService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FInterviewerService.java)|0%|:x:|
|[CoordinatorController.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/f61ca19221b82ea8aa32acd1c0adcb89e1adce1c/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2FCoordinatorController.java)|0%|:x:|

|Total Project Coverage|20.91%|:green_apple:|
|:-|:-:|:-:|

**Comment by knurff** [comment]
|File|Coverage [40.29%]|:green_apple:|
|:-|:-:|:-:|
|[BookingService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FBookingService.java)|100%|:green_apple:|
|[CandidateService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateService.java)|84.03%|:green_apple:|
|[DateUtils.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2FDateUtils.java)|77.01%|:green_apple:|
|[InterviewerService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FInterviewerService.java)|39.53%|:green_apple:|
|[CoordinatorService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCoordinatorService.java)|38.71%|:green_apple:|
|[InterviewerSlot.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2FInterviewerSlot.java)|25.5%|:green_apple:|
|[DashboardDayDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDayDto.java)|13.82%|:green_apple:|
|[DashboardDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDto.java)|9.13%|:x:|
|[CoordinatorController.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/d5d1e5f1222e1c4b1182a9ac05406fa77aaa41db/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2FCoordinatorController.java)|0%|:x:|

|Total Project Coverage|25.3%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java [hunk 1]
```diff
@@ -32,12 +36,12 @@ private DateUtils() {}
    */
   public static int getWeekFrom(LocalDate from, LocalDate current) {
```
Comment (ram333n) [inline src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java:37]: В javadoc цього методу(а саме в `@return`) варто замінити `(1-INT_MAX)` на `(0-INT_MAX)`

File: src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java [hunk 2]
```diff
@@ -67,7 +71,7 @@ public static int getCurrentWeek() {
    * @return the next week number(2-INT_MAX)
```
Comment (ram333n) [inline src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java:71]: Тут варто замінити `(2-INT_MAX)` на `(1-INT_MAX)`

File: src/main/java/com/intellias/intellistart/interviewplanning/service/BookingService.java [hunk 3]
```diff
@@ -1,16 +1,84 @@
...
+import java.util.Set;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
 /**
  * BookingService service.
  */
 @Service
+@AllArgsConstructor
 public class BookingService {
```
Comment (andriimanzar) [inline src/main/java/com/intellias/intellistart/interviewplanning/service/BookingService.java:22]: Методи BookingService я б реалізував через Stream API, виходить значно краще

**Comment by knurff** [comment]
|File|Coverage [37.02%]|:green_apple:|
|:-|:-:|:-:|
|[BookingService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FBookingService.java)|100%|:green_apple:|
|[CandidateService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCandidateService.java)|84.03%|:green_apple:|
|[DateUtils.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Futil%2FDateUtils.java)|77.01%|:green_apple:|
|[InterviewerService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FInterviewerService.java)|39.53%|:green_apple:|
|[CoordinatorService.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fservice%2FCoordinatorService.java)|38.71%|:green_apple:|
|[InterviewerSlot.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fmodel%2FInterviewerSlot.java)|25.5%|:green_apple:|
|[DashboardDayDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDayDto.java)|13.82%|:green_apple:|
|[DashboardDto.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2Fdto%2FDashboardDto.java)|9.13%|:x:|
|[CoordinatorController.java](https://github.com/knurff/intellistart-java-2022-NPE_HUNTERS/blob/39d43c5cfbd3be72aab78eeada3704608ad7592a/src%2Fmain%2Fjava%2Fcom%2Fintellias%2Fintellistart%2Finterviewplanning%2Fcontroller%2FCoordinatorController.java)|0%|:x:|

|Total Project Coverage|23.83%|:green_apple:|
|:-|:-:|:-:|

File: src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java (hunk 1 above)
Comment (Famezzs) [reply src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java:37]: Виконано.

File: src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java (hunk 2 above)
Comment (Famezzs) [reply src/main/java/com/intellias/intellistart/interviewplanning/util/DateUtils.java]: Виконано.

File: src/main/java/com/intellias/intellistart/interviewplanning/service/BookingService.java (hunk 3 above)
Comment (Famezzs) [reply src/main/java/com/intellias/intellistart/interviewplanning/service/BookingService.java:24]: Виконано.

**Review by andriimanzar — APPROVED** [review]
Nice work!

**Pull request merged**